tools/cli/internal/openapi/filter/hidden_envs.go

```diff
@@ -0,0 +1,142 @@
+		if isOperationHiddenForEnv := isOperationHiddenForEnv(operation, metadata); isOperationHiddenForEnv {
+			log.Printf("Removing operation: '%s' from path: '%s' because is hidden for target env: %s", k, pathItem.Ref, metadata.targetEnv)
+			pathItem.SetOperation(k, nil) // Remove Operation if it is hidden for the target environment
+			continue
+		} else if operation.Extensions != nil {
+			// Remove the Hidden extension from the final OAS
+			delete(operation.Extensions, hiddenEnvsExtensionName)
+		}
+
+		for k, response := range operation.Responses.Map() {
+			if isResponseHiddenForEnv := isResponseHiddenForEnv(response, metadata); isResponseHiddenForEnv {
+				log.Printf("Removing response: '%s' from operationID: '%s' because is hidden for target env: %s", k, operation.OperationID, metadata.targetEnv)
+				operation.Responses.Delete(k) // Remove Response if it is hidden for the target environment
+			} else if response.Extensions != nil {
+				// Remove the Hidden extension from the final OAS
+				delete(response.Extensions, hiddenEnvsExtensionName)
+			}
+		}
+
+		if isRequestBodyHiddenForEnv := isRequestBodyHiddenForEnv(operation.RequestBody, metadata); isRequestBodyHiddenForEnv {
+			log.Printf("Removing requestBody from operationID: '%s' because is hidden for target env: %s", operation.OperationID, metadata.targetEnv)
+			operation.RequestBody = nil // Remove RequestBody if it is hidden for the target environment
+		} else if operation.RequestBody != nil && operation.RequestBody.Extensions != nil {
+			// Remove the Hidden extension from the final OAS
+			delete(operation.RequestBody.Extensions, hiddenEnvsExtensionName)
+		}
+	}
+
+	return nil
+}
+
+func isOperationHiddenForEnv(operation *openapi3.Operation, metadata *Metadata) bool {
+	if operation == nil {
+		return false
+	}
+
+	for k, extension := range operation.Extensions {
+		if k != hiddenEnvsExtensionName {
+			continue
+		}
+
+		log.Printf("Found x-hidden-envs: K: %s, V: %s", k, extension)
+		return isHiddenExtensionEqualToTargetEnv(extension, metadata)
+	}
+
+	return false
+}
+
+func isResponseHiddenForEnv(response *openapi3.ResponseRef, metadata *Metadata) bool {
+	if response == nil {
+		return false
+	}
+
+	for k, extension := range response.Extensions {
+		if k != hiddenEnvsExtensionName {
+			continue
+		}
+
+		log.Printf("Found x-hidden-envs in the response: K: %s, V: %s", k, extension)
+		return isHiddenExtensionEqualToTargetEnv(extension, metadata)
+	}
+
+	if response.Value != nil {
+		for k, extension := range response.Value.Extensions {
+			if k != hiddenEnvsExtensionName {
+				continue
+			}
+
+			log.Printf("Found x-hidden-envs in the response: K: %s, V: %s", k, extension)
+			return isHiddenExtensionEqualToTargetEnv(extension, metadata)
+		}
+	}
+
+	return false
+}
+
+func isRequestBodyHiddenForEnv(requestBody *openapi3.RequestBodyRef, metadata *Metadata) bool {
+	if requestBody == nil {
+		return false
+	}
+
+	for k, extension := range requestBody.Extensions {
```

Comment:
`k` tells me is a key of a map, and if a map wouldn't 
```go
v, ok := requestBody.Extensions[hiddenEnvsExtensionName]
if ok {
```
save you the loop?

Reply:
good point, updated